src/realm/sync/noinst/client_impl_base.hpp

```diff
@@ -1336,8 +1359,8 @@ inline void ClientImpl::Session::recognize_sync_version(version_type version)
     if (REALM_LIKELY(resume_upload)) {
         // Since the deactivation process has not been initiated, the UNBIND
         // message cannot have been sent unless an ERROR message was received.
-        REALM_ASSERT(m_error_message_received || !m_unbind_message_sent);
-        if (m_ident_message_sent && !m_error_message_received)
+        REALM_ASSERT(m_suspended || !m_unbind_message_sent);
```

Comment:
should we document somewhere that receiving an error suspends the connection? (the behavior is unchanged, but the new variable does not convey that)

Reply:
You could update the comment at line 1388 as well